src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/Controllers/BalancesController.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.SmartContracts.CLR;
+using Stratis.SmartContracts.Core.Util;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.ReflectionExecutor.Controllers
```

Comment:
How come this is under smart contracts's set of controllers? 

Perhaps we can also put this with the other address balance controller methods, i.e. in the blockstore controller.

Reply:
Basically it's just bringing in this PR from the old 3.0.9.0 branch:

https://github.com/stratisproject/StratisBitcoinFullNode/pull/4186